MozillaFoundation/donate-wagtail
# added apple pay domain association view

## Description

Related PRs/issues https://github.com/MozillaFoundation/foundation.mozilla.org/issues/10262

**This PR adds the following:**
- A new environment variable `APPLE_PAY_DOMAIN_ASSOCIATION_KEY`
- A new view, `apple_pay_domain_association_view` which returns the contents of the env variable when the url `/.well-known/apple-developer-merchantid-domain-association` is visited
- Tests to make sure that the view returns the appropriate response with and without a key set


## Checklist

**Tests**


## Steps to test:

1. Checkout this branch and run `docker-compose up`
2. Visit http://localhost:8000/.well-known/apple-developer-merchantid-domain-association, the page should return the message `Key not found. Please check environment variables.`
3.  Add a test string for the env variable `APPLE_PAY_DOMAIN_ASSOCIATION_KEY`
4. Restart the app
5. Visit http://localhost:8000/.well-known/apple-developer-merchantid-domain-association again, this time, the page should return whatever string you set for the env variable.
6. If everything is working as expected, testing is complete!
